osquery/tables/system/darwin/connected_displays.mm

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#import <AppKit/NSDocument.h>
+#import <Foundation/Foundation.h>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+
+@interface SPDocument : NSDocument {
+}
+- (id)reportForDataType:(id)arg1;
+@end
+
+namespace osquery {
+namespace tables {
+
+QueryData genConnectedDisplays(QueryContext& context) {
+  QueryData results;
+  Row r;
+
+  // BEWARE: Because of the dynamic nature of the calls in this function, we
+  // must be careful to properly clean up the memory. Any future modifications
+  // to this function should attempt to ensure there are no leaks.
```

Comment:
Do autorelease groups help? (I don't know enough to know if they can apply here)